browser/brave_rewards/test/creator_detection_browsertest.cc

```diff
@@ -0,0 +1,360 @@
+  SetRewardsEnabled();
+
+  SetResponseHTML(R"(
+    <!doctype html>
+    <html>
+    <head>
+      <script type="application/ld+json">
+        [
+          {
+            "@type": "Person",
+            "identifier": "987654321",
+            "name": "Test User",
+            "url": "https://vimeo.com/testuser",
+            "image": "https://vimeo.com/user-avatar"
+          }
+        ]
+      </script>
+    </head>
+    <body>
+    </body>
+    </html>
+  )");
+
+  AddBlankTabAndShow(browser());
+  base::test::TestFuture<std::string> future;
+  TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+  ASSERT_TRUE(NavigateTo("vimeo.com", "/testuser"));
+
+  EXPECT_EQ(future.Get<0>(), "vimeo#channel:987654321");
+}
+
+constexpr std::string_view kYouTubeHTML = R"(
+  <!doctype html>
+  <html>
+  <head>
+    <link rel="canonical" href="/channel/987654321" />
+    <script>
+      function triggerSameDocNav() {
+        history.pushState(null, '', '/@testuser2')
+        document.querySelector('#test-owner-link').href = '/@testuser2'
+        document.querySelector('#avatar img').src = '/user-avatar-2'
+        document.querySelector('.ytp-ce-channel-title').href =
+          '/channel/123456789'
+      }
+    </script>
+  </head>
+  <body>
+    <ytd-video-owner-renderer>
+      <a id="test-owner-link" href="/@testuser"></a>
+    </ytd-video-owner-renderer>
+    <div id="avatar">
+      <img id="test-image" src="/user-avatar" />
+    </div>
+    <a class='ytp-ce-channel-title'></a>
+  </body>
+  </html>
+)";
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, YouTubeDetection) {
+  SetRewardsEnabled();
+  SetResponseHTML(kYouTubeHTML);
+
+  AddBlankTabAndShow(browser());
+
+  {
+    base::test::TestFuture<std::string> future;
+    TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+    ASSERT_TRUE(NavigateTo("www.youtube.com", "/@testuser"));
+    EXPECT_EQ(future.Get<0>(), "youtube#channel:987654321");
+  }
+
+  {
+    base::test::TestFuture<std::string> future;
+    TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+    auto* web_contents = chrome_test_utils::GetActiveWebContents(this);
+    EXPECT_TRUE(content::ExecJs(web_contents, "triggerSameDocNav()"));
+    EXPECT_EQ(future.Get<0>(), "youtube#channel:123456789");
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, InvalidHost) {
+  SetRewardsEnabled();
+  SetResponseHTML(kYouTubeHTML);
+  AddBlankTabAndShow(browser());
+  ASSERT_TRUE(NavigateTo("abc.youtube.com", "/@testuser"));
+  WaitForTimeout();
+  EXPECT_EQ(GetRewardsTabHelper().GetPublisherIdForTab(), "");
+}
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, RewardsDisabled) {
+  SetResponseHTML(kYouTubeHTML);
+  AddBlankTabAndShow(browser());
+  ASSERT_TRUE(NavigateTo("www.youtube.com", "/@testuser"));
+  WaitForTimeout();
+  EXPECT_EQ(GetRewardsTabHelper().GetPublisherIdForTab(), "");
```

Comment:
other things happen in RewardsTabHelper like `MaybeSavePublisherInfo` that we don't appear to be testing anywhere.

Reply:
sorry, I guess `MaybeSavePublisherInfo` was already in there so hopefully it's tested somewhere else, but it doesn't look like we're testing that this stuff happens https://github.com/brave/brave-core/pull/25949/files#diff-8126a7ac8b32edde8721529071011a4f05cbb82024ec55561f8af093eed86912R246

Reply:
Tests added for ensuring that the data is stored to the database (see cases `YouTubeDetection` and `NonPlatformSite`).